e2e_test/README.md

```diff
@@ -0,0 +1,94 @@
+* `delegation_verify` app to extract state hashes from generated
+  blocks
+
+Both these apps are handled automatically by the provided `gen_blocks.sh`
+script (they must be compiled manually, though). For the script to work
+the following env variables should be set:
+
+* `DFELEGATION_VERIFY` - the path to the stateless verifier binary
+* `DUMP_BLOCKS` - the path to the dump blocks tool.
+* `BLOCK_DIR` - the path to the directory where the generated blocks will be output
+* `SUBMISSION` - the path to the dummy submission file
+
+The script can be used as follows:
+
+    $ BLOCK_DIR=/home/user/blocks ./gen_block.sh <number of blocks to generate>
+    
+If source code for Mina is present in the file system, `DELEGATION_VERIFY`
+and `DUMP_BLOCKS` paths can be replaced with `MINA_DIR` containing a path
+to the root of the source code repository. Required apps still have to be
+compiled by hand.
+    
+Step 2. Upload
+--------------
+
+Pre-generated blocks should be uploaded to s3, where the mock will be able
+to access them. An example package of 120 consecutive blocks can be found
+here: https://s3.console.aws.amazon.com/s3/buckets/673156464838-mina-precomputed-blocks?region=us-west-2&bucketType=general&prefix=uptime-service-e2e-test/.
+
+The folder should contain any number of `*.dat` files, each of which
+should contain a binary block, and whose name should consist of the
+block's state hash and `.dat` filename extension. Additionally, the
+directory should also contain a special file `block_list.txt`, which
+lists all the block state hashes in order, that is each block on the list
+is the parent of the block on the next line. This file is used by the
+mock to submit blocks in the right order, so that the uptime service
+can reconstruct the dummy blockchain correctly and award uptime points
+accordingly.
+
+Step 3. Running the mock
+------------------------
+
+This step requires an uptime-service already running. It should be configured
+such that it does not verify signatures on submissions. We want to test the
+logic of the uptime service, not signature verification. The mock is run
+as follows:
+
+    $ python generate_submissions.py --block-s3-bucket <bucket> \
+        --block-s3-dir <folder name> <uptime service URL>
+    
+The mock contains a list of 15 hard-coded public keys, which serve as
+block producers' addresses. It loads the list of blocks at the given
+address and forms the chain of state hashes. It also rotates the list
+of block producers so that it never ends. It sets the block pointer to
+the first block on the list.
+
+Then every minute it picks up the next node and sends a submission
+with the block currently pointed to by the block pointer. Every 3 minutes
+it also moves the block pointer to the next block. This way every block
+producer appears to make a submission every 15 minutes and each block
+gets submitted by 3 distinct block producers.
+
+These parameters can be tweaked using command line parameters:
```

Comment:
Formally, there is none. In practice if you set the submission time too low, the script won't keep up and you'll get submissions sent as fast as your machine can handle. Block time has no limit at all, but in practice it makes no sense to have it lower than the submission time, since this is when the script checks whether or not it's time to switch to the next block. It will never skip blocks, even if the block time is so low that it theoretically should. Let me add these explanations to the README.